### PR TITLE
[feature] new img container support PNG

### PR DESCRIPTION
**Pull request by Hanyonggong**
New img container support PNG based on libpng

* Supports PNG images in all formats
* Maintain the code style of the new container (like BMP support) in order to abandon libpng in the future
* It have passed the Lenet model test

To compile target buddy-new-image-container-test:

$ cd buddy-mlir
$ mkdir build
$ cd build
$ cmake -G Ninja .. \
    -DMLIR_DIR=$PWD/../llvm/build/lib/cmake/mlir \
    -DLLVM_DIR=$PWD/../llvm/build/lib/cmake/llvm \
    -DLLVM_ENABLE_ASSERTIONS=ON \
    -DCMAKE_BUILD_TYPE=RELEASE
    -DBUDDY_MLIR_ENABLE_DIP_LIB=ON
$ ninja

### CODE REVIEW

**Comment by LiuYang328** [comment]
Because of the new dependencies and changes to the cmake file, a new version README is needed to guide users build with new container

**Comment by Hanyonggong** [comment]
> Because of the new dependencies and changes to the cmake file, a new version README is needed to guide users build with new container

The build commands for the examples has not changed, but the build command for test has changed

**Comment by LiuYang328** [comment]
> > Because of the new dependencies and changes to the cmake file, a new version README is needed to guide users build with new container
> 
> The build commands for the examples has not changed, but the build command for test has changed

when i build buddy-mlir, i have to enable BUDDY_MLIR_ENABLE_DIP_LIB, otherwise, the build won't succeed. It's best to explain this in the pr.

**Comment by Hanyonggong** [comment]
> when i build buddy-mlir, i have to enable BUDDY_MLIR_ENABLE_DIP_LIB, otherwise, the build won't succeed. It's best to explain this in the pr.

Thank you for your question. I have updated the PR to prevent compilation failure due to unspecified parameters

**Comment by LiuYang328** [comment]
Does libpng have specific version requirements?  My current version is 1.6.37
image

when i use new container in MobileNet example, I meet this error

image

Can you try the MobileNet example on your side ?  Thank you !

**Comment by Hanyonggong** [comment]
> Does libpng have specific version requirements? My current version is 1.6.37 image
> 
> when i use new container in MobileNet example, I meet this error
> 
> image
> 
> Can you try the MobileNet example on your side ? Thank you !

Please refer to the Lenet's CMakeLists.txt for the correct link to the libpng library

File: frontend/Interfaces/buddy/DIP/ImgContainer.h [hunk 1]
```diff
@@ -390,6 +413,199 @@ bool Image<T, N>::decodeBMP(const std::vector<uint8_t> &fileData) {
...
+        }
+    } else if (this->imageMode == ImageModes::DIP_RGB) {
+      // TODO: Add batch setting.
+      this->sizes[0] = 1;
+      this->sizes[1] = 3;
+      this->sizes[2] = this->height;
+      this->sizes[3] = this->width;
+      this->setStrides();
+      size_t size = this->product(this->sizes);
+      std::cout << "size" << size << std::endl;
```
Comment (LiuYang328) [inline frontend/Interfaces/buddy/DIP/ImgContainer.h:551]: cout 删除

**Pull request merged**